### PR TITLE
Typescript nvim-lsp setup

### PR DESCRIPTION
**Pull request by dbalatero**
TODO:

- [x] Figure out why I hardcode the PATH for the binary
- [ ] Speed up completion results?

### CODE REVIEW

File: nvim/lua/lspservers.lua [hunk 1]
```diff
@@ -295,6 +295,25 @@ lspconfig.emmet_ls.setup({
   capabilities = lspCapabilities,
 })
 
+lspconfig.tsserver.setup({
+  capabilities = lspCapabilities,
+  -- This sucks - npm install -g typescript-language-server is not able to be seen
+  -- by my $PATH, so I just hard code the path here for my laptop.
+  cmd = {'/Users/dbalatero/.nodenv/versions/14.18.1/bin/typescript-language-server', '--stdio'},
```
Comment (dbalatero) [inline nvim/lua/lspservers.lua]: @sodiumjoe I notice [your config](https://github.com/sodiumjoe/dotfiles/blob/master/neovim/lua/sodium/plugins.lua#L274-L314) didn't require you to pass in the full path to the binary, but my neovim is not resolving the `PATH` to this binary correctly.

I did an `npm install -g typescript-language-server` and I manage node with nodenv. Is there something special I have to do to get my PATH to recognize `npm install -g` modules?

File: nvim/lua/lspservers.lua [hunk 2]
```diff
@@ -295,6 +295,25 @@ lspconfig.emmet_ls.setup({
...
+  cmd_env = { NODE_OPTIONS = "--max-old-space-size=8192" }, -- Give 8gb of RAM to node
+  filetypes = { "typescript", "typescriptreact", "typescript.tsx" },
+  init_options = {
+    maxTsServerMemory = "8192",
+  },
+  root_dir = lspconfig.util.root_pattern("tsconfig.json"),
+  on_attach = function(client, bufnr)
+    client.resolved_capabilities.document_formatting = false
+    client.resolved_capabilities.document_range_formatting = false
```
Comment (dbalatero) [inline nvim/lua/lspservers.lua:311]: @moon idk how much it matters but I noticed your config didn't have this `document_range_formatting` thing set to `false` which I cribbed from here:

https://jose-elias-alvarez.medium.com/configuring-neovims-lsp-client-for-typescript-development-5789d58ea9c

I kind of read it as maybe… you can visual select a block of code and format only that?

File: nvim/lua/lspservers.lua (hunk 1 above)
Comment (sodiumjoe) [reply nvim/lua/lspservers.lua]: Not sure, I don't think I have anything special to get that in my path. Does running `typescript-language-server` from your shell work?

relevant stuff for me:
```
❯ which typescript-language-server
/usr/local/bin/typescript-language-server
❯ ll $(which typescript-language-server)
{snip} /usr/local/bin/typescript-language-server -> {~}/.config/yarn/global/node_modules/.bin/typescript-language-server
```

Comment (dbalatero) [reply nvim/lua/lspservers.lua]: > Not sure, I don't think I have anything special to get that in my path. Does running typescript-language-server from your shell work?

No, it's stuck inside of that versioned nodenv npm path.

```
/usr/local/bin/typescript-language-server -> {~}/.config/yarn/global/node_modules/.bin/typescript-language-server
```

Ah, I bet you ran `yarn global add typescript-language-server` in that case?

Comment (dbalatero) [reply nvim/lua/lspservers.lua]: `yarn global` fixed it for me!

Comment (sodiumjoe) [reply nvim/lua/lspservers.lua]: ah, yeah that needs to be global

**Pull request merged**